### ISSUE TITLE
Availability of emitOnAllTabs props in the useIdleTimer hook - upcoming V5

### ISSUE THREAD

**Issue by honghainguyen777** [SupremeTechnopriest/react-idle-timer#243]
## Question

One of the features I love about the `react-idle-timer` is the `crossTab`. In the `4.6.4` version, there is `emitOnAllTabs` for emitting the `onIdle` and `onActive` functions on all tabs.
```js
crossTab: {
   emitOnAllTabs: true,
}
```

I often have some actions that emit for all tabs and some that execute in the lead tab only (with `crossTab: true`). However, the new V5 (ie. 5.0.0-rc.24 currently) has no `emitOnAllTabs` but `crossTab: boolean` only (as I know so far from the type definitions).

Could you guys add the `emitOnAllTabs` back to the V5?

Side question: Can I have two `useIdleTimer` hooks in a component? Some actions I want to emit on all tabs and some for the lead tab only.

Thank you!

**Comment by SupremeTechnopriest** [comment]
Hi @honghainguyen777,

So in v5 `emitOnAllTabs` is the default behavior. So no worries there you wont loose it. However, in v5 leader election was removed.  It was incompatible with new features like the prompt.  I exposed the message channel via the `onMessage` prop and the `message` method, so you could implement it yourself.  You can have two `useIdleTimer` hooks if you like.  There should be no issues there, but I dont think you need to.  You could just do the leader filter inside your callbacks.  For example:

```js
const App = () => {
  const [isLeader, setIsLeader] = useState(false)
  
  const onMessage = (message) => {
    // Another window is leader
    // Disable this window
    if (message.leader) {
      setIsLeader(false)
    }
  }

  const onIdle = () => {
    if (isLeader) {
      // Do something if I am the leader
    } else {
      // Do something if I am not the leader
    }
  }

  const onActive = () => {
    if (isLeader) {
      // Do something if I am the leader
    } else {
      // Do something if I am not the leader
    }
  }

  const idleTimer = useIdleTImer({
    onIdle,
    onActive,
    onMessage
    crossTab
  })

  // Determine the leader how ever you want
  // This is example sets the leader to the
  // visible window
  useEffect(() => {
    const onPageHide = () => {
      setIsLeader(false)
      idleTimer.message({ leader: false })
    }
    const onPageShow = () => {
      setIsLeader(true)
      idleTimer.message({ leader: true })
    }

    window.addEventListener('pagehide', onPageHide)
    window.addEventListener('pageshow', onPageShow)

    return () => {
      window.removeEventListener('pagehide', onPageHide)
      window.removeEventListener('pageshow', onPageShow)
    }
  })
```

This is an (untested) example of how you might implement leader election. The v4 leader election didn't take into account the visible tab. It just selected the first tab to come online and made that tab the leader until it was either closed or became unresponsive. Hope this helps!

**Comment by honghainguyen777** [comment]
Hi @SupremeTechnopriest , I hope that you are doing great! I will implement the leader election on my project and let you know here if it works. Thank you so much for your quick response and cool example!

**Comment by honghainguyen777** [comment]
Hi @SupremeTechnopriest It works! Thank you so much!

I have to spend some time working around the case if the leader is closed to pass the leadership to another tab.
```js
window.addEventListener('visibilitychange', onVisibilityChange); -> if the visibility changes -> give the leadership to the active tab
window.addEventListener('beforeunload', onGivingUpLeadership); -> if the lead tab is closed -> announce to other tabs to take over the leadership (or remove `leader` item from the localStorage)
localStorage.setItem('leader', '1'); -> make sure that the leadership has been taken over. If not (`leader = null`), ask one of other tabs to take it (set the `leader` to `1` and announce to the other tab that the leadership has been taken)
```
Have a great weekend!

**Comment by SupremeTechnopriest** [comment]
Glad I could help!  Going to close this for now.  Feel free to re open or pop in discord if you need more help!

**Comment by SupremeTechnopriest** [comment]
Hey @honghainguyen777!

I just published `5.1.0` which brings back Leader Election. There are some subtle differences from v4 that are outlined [here](https://idletimer.dev/docs/getting-started/new#leader-election). Let me know how it goes!
